test/query_component_test.dart

```diff
@@ -58,156 +59,333 @@ class OverrideDirective {}
+            .thenAnswer((_) => Future.value(reportingGetQueryApiJsonResponse));
+        when(mockQueryService.execReportingDownload(downloadLink))
+            .thenAnswer((_) => Future.value(generateReport(report)));
+
+        expectedInput = [
+          generateInsertionOrder(
+              advertiserId, mediaPlanIdA, ioIdA, startDate, endDate, spendingA),
+          generateInsertionOrder(
+              advertiserId, mediaPlanIdA, ioIdB, startDate, endDate, spendingB),
+          generateInsertionOrder(
+              advertiserId, mediaPlanIdA, ioIdC, startDate, endDate, spendingC),
+        ];
+      });
+
+      test(
+          'highlight underpacing unchecked invokes populate(multiple-IO, false)',
+          () async {
+        await queryComponentPO.clickPopulate();
+
+        await Future.delayed(Duration(seconds: 2));
+        verify(mockExcelDart.populate(expectedInput, false));
+      });
+
+      test('highlight underpacing checked invokes populate(multiple-IO, true)',
+          () async {
+        await queryComponentPO.selectUnderpacing();
+        await queryComponentPO.clickPopulate();
+
+        await Future.delayed(Duration(seconds: 2));
+        verify(mockExcelDart.populate(expectedInput, true));
+      });
+    });
+
+    group('selecting ByMediaPlan with', () {
+      List<InsertionOrder> expectedInput;
+
+      setUp(() async {
+        await queryComponentRadioButtonPO.selectByIO();
+        await queryComponentPO.typeAdvertiserId(advertiserId);
+        await queryComponentPO.typeInsertionOrderId(ioIdA);
+
+        final singleIOJsonString =
+            '${generateInsertionOrderJsonString(advertiserId, mediaPlanIdA, ioIdA, startDate, endDate)}';
+        final report = '$ioIdA,2020/07/10,$spendingA,1000\\n';
+
+        when(mockQueryService.execDV3Query(QueryType.byInsertionOrder, '',
+                advertiserId, argThat(isNull), ioIdA))
+            .thenAnswer((_) => Future.value(singleIOJsonString));
+        when(mockQueryService.execReportingCreateQuery(
+                QueryType.byInsertionOrder,
+                advertiserId,
+                argThat(isNull),
+                ioIdA,
+                startDate,
+                any))
+            .thenAnswer(
+                (_) => Future.value(reportingCreateQueryApiJsonResponse));
+        when(mockQueryService.execReportingGetQuery(queryId))
+            .thenAnswer((_) => Future.value(reportingGetQueryApiJsonResponse));
+        when(mockQueryService.execReportingDownload(downloadLink))
+            .thenAnswer((_) => Future.value(generateReport(report)));
+
+        expectedInput = [
+          generateInsertionOrder(
+              advertiserId, mediaPlanIdA, ioIdA, startDate, endDate, spendingA)
+        ];
+      });
+
+      test('highlight underpacing unchecked invokes populate(single-IO, false)',
+          () async {
+        await queryComponentPO.clickPopulate();
+
+        await Future.delayed(Duration(seconds: 2));
```

Comment:
drop a comment by these futures to indicate why they are here. normally we try to avoid these (even though theyre necessary here) so an explanation may be necessary